lib/open_food_network/scope_variants_for_search.rb

```diff
@@ -68,6 +69,18 @@ def scope_to_eligible_for_subscriptions_in_distributor
       scope_variants_to_distributor(@variants, distributor)
     end
 
+    def scope_to_in_stock_only
+      @variants = @variants.joins(
+        "INNER JOIN spree_stock_items ON spree_stock_items.variant_id = spree_variants.id
+         LEFT JOIN variant_overrides ON variant_overrides.variant_id = spree_variants.id"
+      ).where("
+        variant_overrides.on_demand IS TRUE OR
+        variant_overrides.count_on_hand > 0 OR
+        (variant_overrides.on_demand IS NULL AND spree_stock_items.backorderable IS TRUE) OR
+        (variant_overrides.count_on_hand IS NULL AND spree_stock_items.count_on_hand > 0)
+      ")
+    end
```

Comment:
Hmm... I think at this point we might need to join only those `variant_overrides` which also match the `distributor_id` we're looking for, otherwise we'll get weird results here because a variant can have many variant overrides across many distributors.

We'll have access to `params[:distributor_id]` when calling this method though, right? So it should be simple.

Reply:
No problem, that's done now. I have a test which covers that scenario too [here](https://github.com/openfoodfoundation/openfoodnetwork/pull/8017/files#diff-5af409bc212eb446a4100e0d3e6029aecaaff06fd3e2e4b774449fb85fcdd94bR146). Although it took me a while it figure out why it wasn't failing before, it looks like the `scope_to_distributor` call prevents this anyway but it's much clearer adding the distributor condition on the left join too.

If you want me to rebase this let me know.